### DISCUSSION TITLE
Packages/com.unity.ugui/api/UnityEngine.EventSystems.IDropHandler

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#262] [Notes]
# Packages/com.unity.ugui/api/UnityEngine.EventSystems.IDropHandler

https://docs.unity3d.com/Packages/com.unity.ugui@latest/index.html?subfolder=/api/UnityEngine.EventSystems.IDropHandler.html

**Comment by radiatoryang** [comment]
If OnDrop doesn't seem to be working or firing, there are two ways to make it work:  
OPTION 1: put a CanvasGroup on the dragged object, and disable "Blocks Raycast" when you begin dragging, and then re-enable it when you end dragging  
OPTION 2: make sure the object accepting the drop is listed AFTER the dragged item in the hierarchy (order matters). That is, your hierarchy order should look like this:  
Hierarchy:  
- dragged item implementing IDragHandler  
- drop area implementing IDropHandler
